core/src/main/scala/org/apache/spark/SparkHadoopWriter.scala

```diff
@@ -106,18 +107,25 @@ class SparkHadoopWriter(@transient jobConf: JobConf)
     val taCtxt = getTaskContext()
     val cmtr = getOutputCommitter()
     if (cmtr.needsTaskCommit(taCtxt)) {
-      try {
-        cmtr.commitTask(taCtxt)
-        logInfo (taID + ": Committed")
-      } catch {
-        case e: IOException => {
-          logError("Error committing the output of task: " + taID.value, e)
-          cmtr.abortTask(taCtxt)
-          throw e
+      val outputCommitCoordinator = SparkEnv.get.outputCommitCoordinator
+      val conf = SparkEnv.get.conf
+      val canCommit: Boolean = outputCommitCoordinator.canCommit(jobID, splitID, attemptID)
+      if (canCommit) {
```

Comment:
Man, github makes it *really* hard to find this discussion. :-/

Anyway, just one comment:
> if the DAGScheduler did not authorize the commit, then we should throw an exception

As long as that doesn't cause the driver's "task error count" go up, then fine. Otherwise, we probably need some new state (e.g. "task gave up") that doesn't count as an error.